pkg/workceptor/kubernetes.go

```diff
@@ -902,12 +1015,49 @@ func (kw *KubeUnit) runWorkUsingLogger() {
 		return
 	}
 
-	// only transition from WorkStateRunning to WorkStateSucceeded if WorkStateFailed is set we do not override
+	pod, err := kw.KubeAPIWrapperInstance.Get(kw.GetContext(), kw.clientset, podNamespace, podName, metav1.GetOptions{})
+	if err != nil {
+		kw.GetWorkceptor().nc.GetLogger().Warning("Failed to retrieve pod for diagnostics: %v", err)
+	} else if false { // deactivate new code TODO write integration tests
+		ok, _ := kw.CapturePodStatus(pod, stdout.Size(), 10*time.Second)
+		if !ok {
+			// If the pod did not succeed, we already updated the status to WorkStateFailed
+			// and we can return early.
+			return
+		}
+	}
+
+	// Only transition to WorkStateSucceeded if the work unit is still running
+	// and has not already failed due to diagnostic or streaming errors.
+	// This ensures we don't override a previously set WorkStateFailed.
 	if kw.GetContext().Err() != context.Canceled && kw.Status().State == WorkStateRunning {
 		kw.UpdateBasicStatus(WorkStateSucceeded, "Finished", stdout.Size())
 	}
 }
 
+func (kw *KubeUnit) CapturePodStatus(pod *corev1.Pod, stdoutSize int64, timeout time.Duration) (ok bool, err error) {
+
+	pod, err = kw.KubeAPIWrapperInstance.WaitForPodCompleted(pod, kw.clientset, timeout)
+	if err != nil {
+		kw.GetWorkceptor().nc.GetLogger().Debug("Job complete and pod is still running: %v", err)
+	}
+
+	ok, reason, err := kw.KubeAPIWrapperInstance.GetPodStatus(pod)
+	if err != nil {
+		reason := fmt.Sprintf("Pod diagnostics failed: %v reason %s", err, reason)
+		kw.GetWorkceptor().nc.GetLogger().Warning(reason)
```

Comment:
```suggestion
		kw.GetWorkceptor().nc.GetLogger().Warning("%s", reason)
```
This will fix the lint issue by providing a constant format string.